### PR TITLE
Ping msg

### PR DESCRIPTION
**Pull request by jwade109**
Adds communication between drive_node and test_controller: drive_node regularly publishes Status messages to /robot/status, and must recieve an Int8 message atleast once every 2 seconds via /driver/ping or it will shut down until it does.

### CODE REVIEW

File: src/robot2017/src/drive_node.cpp [hunk 1]
```diff
@@ -41,31 +66,47 @@ int main(int argc, char **argv)
...
-    ros::Subscriber sub_tele = nh.subscribe("/robot/teleop", 1000, &RobotExec::teleopReceived, &exec);
-
-    ros::Subscriber sub_aut = nh.subscribe("/robot/autonomy", 1000, &RobotExec::autonomyReceived, &exec);
-
-    ros::Publisher pub_fb = nh.advertise<robot_msgs::MotorFeedback>("/robot/autonomy/feedback", 100);
+    pub_fb = nh.advertise<robot_msgs::MotorFeedback>("/robot/autonomy/feedback", 100);
 
+    pub_status = nh.advertise<robot_msgs::Status>("/robot/status", 100);
+    driver_ping = nh.subscribe("/driver/ping", 1000, &recievedPing);
+    ros::Publisher fake_ping = nh.advertise<std_msgs::Bool>("/driver/ping", 100);
```
Comment (anidev) [inline src/robot2017/src/drive_node.cpp]: Doesn't look like this is necessary anymore

File: src/robot2017/src/drive_node.cpp [hunk 2]
```diff
@@ -41,31 +66,47 @@ int main(int argc, char **argv)
...
 
     RobotExec exec(onPC, debug, autoEnable);
 
-    // establish this program as an ROS node.
-    ros::NodeHandle nh;
-
-    ros::Subscriber sub_tele = nh.subscribe("/robot/teleop", 1000, &RobotExec::teleopReceived, &exec);
-
-    ros::Subscriber sub_aut = nh.subscribe("/robot/autonomy", 1000, &RobotExec::autonomyReceived, &exec);
-
```
Comment (anidev) [inline src/robot2017/src/drive_node.cpp]: You also removed the subscriptions for teleop and autonomy - should add them back

File: src/robot2017/src/drive_node.cpp [hunk 3]
```diff
@@ -41,31 +66,47 @@ int main(int argc, char **argv)
...
     int hertz = 100;
     ros::Rate r(hertz);
 
     while(ros::ok())
     {
         ros::spinOnce();
 
-        if (exec.isAutonomyActive())
+        double timeSince = (ros::Time::now() - lastDriverPing).toSec();
+        if (timeSince > 2 && !hibernating)
```
Comment (anidev) [inline src/robot2017/src/drive_node.cpp]: Could you replace the 2 with a constant int at the top of the file

File: src/robot2017/src/drive_node.cpp [hunk 4]
```diff
@@ -74,9 +115,8 @@ int main(int argc, char **argv)
             pub_fb.publish(motorFb);
         }
 
+        publishStatus();
```
Comment (anidev) [inline src/robot2017/src/drive_node.cpp]: It doesn't look you're actually setting the values for the status message anywhere before publishing

**Review by anidev — CHANGES_REQUESTED** [review]
Why did you change the launch files - they should probably be changed back, since each launch file is meant to be standalone, so the test launch needs to launch both drive_node as well as test. Also, the main launch file shouldn't have `-pc -debug -aut`, those are just for testing.

File: src/robot2017/src/drive_node.cpp (hunk 2 above)
Comment (jwade109) [reply src/robot2017/src/drive_node.cpp]: resolved. not sure why I removed them.

File: src/robot2017/src/drive_node.cpp (hunk 1 above)
Comment (jwade109) [reply src/robot2017/src/drive_node.cpp]: correct, that was just for testing. removed

File: src/robot2017/src/drive_node.cpp (hunk 4 above)
Comment (jwade109) [reply src/robot2017/src/drive_node.cpp]: not sure where you want it to be set. I figured there would be something in the ros::ok() loop.

**Review by anidev — APPROVED** [review]

**Comment by anidev** [comment]
Finally been merged into branch bbb_hw_libraries

**Pull request merged**